api/features/views.py

```diff
@@ -110,10 +111,19 @@ def perform_update(self, serializer):
         serializer.save(project_id=self.kwargs.get("project_pk"))
 
     def perform_destroy(self, instance):
-        self._create_feature_delete_audit_log(instance)
+        feature_states = instance.feature_states.filter(
+            identity=None, feature_segment=None
+        )
+        self._create_feature_delete_audit_log(feature_states)
+        self._trigger_feature_state_change_webhooks(feature_states)
```

Comment:
Could we not handle this on the historical feature state post_save signal? 

I'm just thinking that this will not catch the deletion of e.g. identity / segment override feature states. Maybe that's not the desired behaviour but might be worth checking.

Reply:
Oh, yes, that was the starting point of this, but the problem with that is that `feature_segment` get deleted before `feature state` and when feature_states tries to fetch feature segment it fails with a violent error.